### ISSUE TITLE
regsort handle the case when no cycle and xchg=False incorrectly.

### ISSUE THREAD

**Issue by peter50216** [Gallopsled/pwntools#847]
```python
>>> from pwn import *
>>> regsort({'a': 'b', 'b': 'c'}, ['a', 'b', 'c'])
[('mov', 'a', 'b'), ('mov', 'b', 'c')]
>>> regsort({'a': 'b', 'b': 'c'}, ['a', 'b', 'c'], xchg=False)
[('mov', 'b', 'c'), ('mov', 'a', 'b')]
```
Both output should be same.

**Comment by zachriggle** [comment]
Holy cow, I'm impressed anybody ever dug into that code! Thanks for
reporting the bug, I'll look tomorrow!

Out of curiosity, we experienced a spike in reports this week -- did
pwntools come up in media somewhere?

**Comment by peter50216** [comment]
For the reports by me or @david942j, it's because we're trying to write pwntools for Ruby :D
(https://github.com/peter50216/pwntools-ruby)

**Comment by zachriggle** [comment]
Fixed via #852
